platform/ext/target/nordic_nrf/common/core/hw_init.c

```diff
@@ -19,6 +26,78 @@ static void hw_init_mpu(void)
 	}
 }
 
+#if defined(NRF_HW_INIT_NRF_PERIPHERALS)
+#if defined(NRF_RTC0) || defined(NRF_RTC1) || defined(NRF_RTC2)
+static inline void nrf_cleanup_rtc(NRF_RTC_Type * rtc_reg)
+{
+    nrf_rtc_task_trigger(rtc_reg, NRF_RTC_TASK_STOP);
+    nrf_rtc_event_disable(rtc_reg, 0xFFFFFFFF);
+    nrf_rtc_int_disable(rtc_reg, 0xFFFFFFFF);
+}
+#endif
+
+static void nrf_cleanup_uarte(NRF_UARTE_Type * uarte_reg)
+{
+/* All subscribe + reserved, i.e from SUBSCRIBE_STARTRX to EVENTS_CTS */
+#define NRF_UARTE_SUBSCRIBE_CONF_OFFS offsetof(NRF_UARTE_Type, SUBSCRIBE_STARTRX)
+#define NRF_UARTE_SUBSCRIBE_CONF_SIZE (offsetof(NRF_UARTE_Type, EVENTS_CTS) -\
+                                       NRF_UARTE_SUBSCRIBE_CONF_OFFS)
+
+/* All publish + reserved, i.e from PUBLISH_CTS to SHORTS */
+#define NRF_UARTE_PUBLISH_CONF_OFFS offsetof(NRF_UARTE_Type, PUBLISH_CTS)
+#define NRF_UARTE_PUBLISH_CONF_SIZE (offsetof(NRF_UARTE_Type, SHORTS) -\
+                                     NRF_UARTE_PUBLISH_CONF_OFFS)
+
+    nrf_uarte_disable(uarte_reg);
+    nrf_uarte_int_disable(uarte_reg, 0xFFFFFFFF);
+#if defined(NRF_DPPIC)
+    /* Clear all SUBSCRIBE configurations. */
+    memset((uint8_t *)uarte_reg + NRF_UARTE_SUBSCRIBE_CONF_OFFS, 0, NRF_UARTE_SUBSCRIBE_CONF_SIZE);
+    /* Clear all PUBLISH configurations. */
+    memset((uint8_t *)uarte_reg + NRF_UARTE_PUBLISH_CONF_OFFS, 0, NRF_UARTE_PUBLISH_CONF_SIZE);
+#endif
+}
+
+static void nrf_cleanup_peripheral_interconnect(void)
+{
+#if defined(NRF_PPI)
+    nrf_ppi_channels_disable_all(NRF_PPI);
+#endif
+#if defined(NRF_DPPIC)
+    nrf_dppi_channels_disable_all(NRF_DPPIC);
+#endif
+}
+
+static void nrf_cleanup_clock(void)
+{
+    nrf_clock_int_disable(NRF_CLOCK, 0xFFFFFFFF);
+}
+
+static void hw_init_nrf_peripherals(void)
+{
+#if defined(NRF_RTC0)
+    nrf_cleanup_rtc(NRF_RTC0);
+#endif
+#if defined(NRF_RTC1)
+    nrf_cleanup_rtc(NRF_RTC1);
+#endif
+#if defined(NRF_RTC2)
+    nrf_cleanup_rtc(NRF_RTC2);
+#endif
+
+#if defined(NRF_UARTE0)
+    nrf_cleanup_uarte(NRF_UARTE0);
+#endif
+#if defined(NRF_UARTE1)
```

Comment:
Generally looks good, I just wonder where these symbols come from? `NRF_UARTE1` etc. 
Is there any mapping between these and what is set for MCUBoot or does this just generally reset whatever hardware is available on the chip?